tests/queries/0_stateless/03705_function_dict_get_keys_multiple_dict_and_no_caching.sql

```diff
@@ -0,0 +1,90 @@
+CREATE TABLE src_products
+(
+    id UInt64,
+    category String,
+    brand String
+)
+ENGINE = Memory;
+
+INSERT INTO src_products VALUES
+    (1, 'catA', 'brandX'),
+    (2, 'catA', 'brandY'),
+    (3, 'catB', 'brandX'),
+    (4, 'catC', 'brandZ'),
+    (5, 'catB', 'brandZ');
+
+CREATE TABLE src_geo
+(
+    country String,
+    city String,
+    timezone String,
+    code UInt32
+)
+ENGINE = Memory;
+
+INSERT INTO src_geo VALUES
+    ('US', 'NYC',     'UTC-5', 10001),
+    ('US', 'Chicago', 'UTC-6', 60601),
+    ('FR', 'Paris',   'UTC+1', 75000),
+    ('DE', 'Berlin',  'UTC+1', 10115),
+    ('JP', 'Tokyo',   'UTC+9', 100000);
+
+CREATE DICTIONARY dict_products
+(
+    id UInt64,
+    category String,
+    brand String
+)
+PRIMARY KEY id
+SOURCE(CLICKHOUSE(TABLE 'src_products'))
+LIFETIME(MIN 0 MAX 0)
+LAYOUT(HASHED());
+
+CREATE DICTIONARY dict_geo
+(
+    country String,
+    city String,
+    timezone String,
+    code UInt32
+)
+PRIMARY KEY country, city
+SOURCE(CLICKHOUSE(TABLE 'src_geo'))
+LIFETIME(MIN 0 MAX 0)
+LAYOUT(COMPLEX_KEY_HASHED());
+
+CREATE TABLE inputs
+(
+    target_category String,
+    target_brand String,
+    target_timezone String
+)
+ENGINE = Memory;
+
+INSERT INTO inputs VALUES
+    ('catA', 'brandX', 'UTC+1'),
+    ('catB', 'brandZ', 'UTC+9'),
+    ('catX', 'brandX', 'UTC-6');
+
+SELECT
+    target_category,
+    target_brand,
+    target_timezone,
+    dictGetKeys('dict_products', 'category', target_category) AS product_ids_by_category,
+    dictGetKeys('dict_products', 'brand',    target_brand)    AS product_ids_by_brand,
+    dictGetKeys('dict_geo',      'timezone', target_timezone) AS country_city_by_tz
```

Comment:
Please also add examples where a composite expression is used for the value argument, e.g., such as `concat(c0, c1)`;